app/scripts/modules/core/src/pipeline/config/templates/v2/configurePipelineTemplateModalV2.controller.ts

```diff
@@ -0,0 +1,251 @@
+import { module, IController, IScope, IHttpPromiseCallbackArg, IPromise } from 'angular';
```

Comment:
Rather than add complexity to the V1 `configurePipelineTemplateModal.controller.ts` and risk breaking support for v1 templates, I have split the v2 logic into its own controller and test files. 

This will make it easier to remove V1 files once the templates are fully deprecated and migrated over. Much of this controller is similar to V1 but I have annotated major differences.